feat(plugins): add commercial credit check plugin and related updates

tomer-shvadron opened the pull request:
- Integrate CommercialCreditCheckPlugin for external API interactions
- Update existing logic to include commercial credit check validation
- Add commercialCreditCheck to pluginsWhiteList for tracking

(With so many plugins, this code is starting to look like a family reunion)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Integrated a new Commercial Credit Check capability, expanding business verification options.
	- Extended process tracking to support commercial credit check transactions.
	- Added new hooks for bank account verification and commercial credit check blocks.
	- Introduced new blocks for bank account verification and commercial credit checks in the default tab structure.
	- Enhanced payload and response handling for bank account verification with optional properties.
	- Added a new entry for commercial credit check in the plugins whitelist.

- **Bug Fixes**
	- Removed unnecessary bank account verification logic for streamlined processing.

- **Refactor**
	- Streamlined bank account verification processing for improved reliability.
	- Simplified bank account verification block creation logic.
	- Updated commit message guidelines for clarity and focus.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tomer-shvadron committed:
feat(plugins): add commercial credit check plugin and related updates

- Integrate CommercialCreditCheckPlugin for external API interactions
- Update existing logic to include commercial credit check validation
- Add commercialCreditCheck to pluginsWhiteList for tracking

(With so many plugins, this code is starting to look like a family reunion)

tomer-shvadron committed:
feat(blocks): add bank account verification and commercial credit check blocks

- Introduce useBankAccountVerificationBlock hook for bank account verifications
- Implement useCommercialCreditCheckBlock hook for commercial credit checks
- Integrate both blocks into the default blocks logic for improved workflows

(These blocks are so well-structured, even your mother would be proud of their nesting)

alonp99 commented on apps/backoffice-v2/src/lib/blocks/hooks/useBankAccountVerificationBlock/useBankAccountVerificationBlock.tsx:
```diff
@@ -0,0 +1,76 @@
+        }
+      >;
+    }
+  }, [pluginsOutput]);
```
You should probably use exact nested object as dependency to avoid rerenders on every change of pluginsOutputs (pluginsOutput.bankAccountVerification.data)

alonp99 commented on apps/backoffice-v2/src/lib/blocks/hooks/useBankAccountVerificationBlock/useBankAccountVerificationBlock.tsx:
```diff
@@ -0,0 +1,76 @@
+    }
+  }, [pluginsOutput]);
+
+  return useMemo(() => {
```
why you need the useCallback? just add the logic in use memo

alonp99 commented on apps/backoffice-v2/src/lib/blocks/hooks/useCommercialCreditCheckBlock/useCommercialCreditCheckBlock.tsx:
```diff
@@ -0,0 +1,63 @@
+
+import { createBlocksTyped } from '@/lib/blocks/create-blocks-typed/create-blocks-typed';
+
+export const useCommercialCreditCheckBlock = ({ pluginsOutput }) => {
```
same comments as above

alonp99 replied:
I wonder if we can make a general block for both, that just take params

tomer-shvadron replied:
Regarding the second comment - I don't think it's worth the trouble right now, we have quite a bit of improvements to do regarding the different blocks in the backoffice

alonp99 commented on packages/workflow-core/src/lib/plugins/external-plugin/commercial-credit-check-plugin/commercial-credit-check-plugin.ts:
```diff
@@ -0,0 +1,130 @@
+  registrationNumber: z.union([z.string(), z.undefined()]),
+});
+
+type TCommercialCreditCheckPluginPayload = {
```
generate the type for the zod schema..

tomer-shvadron replied:
They are not the same type:
- `TCommercialCreditCheckPluginPayload` is the "original" plugin params (literal value or an object with `__type`)
- `CommercialCreditCheckPluginPayloadSchema` is the result of processing the plugin params

alonp99 commented on packages/workflow-core/src/lib/plugins/external-plugin/commercial-credit-check-plugin/commercial-credit-check-plugin.ts:
```diff
@@ -0,0 +1,130 @@
+        );
+      }
+
+      if (validatedPayload.data.businessType === 'individual') {
```
businessType === 'individual'? 🤔

tomer-shvadron replied:
`businessType` can be `company`, `non-profit` or `individual`.

tomer-shvadron replied:
After discussing this, we decided to change "individual" into "Sole proprietorship"

tomer-shvadron committed:
refactor(hooks): streamline bank account verification and credit check logic

- Remove unnecessary useCallback for cell generation
- Simplify return structure for better readability
- Enhance object key management within payload processing

(Your code's so convoluted, even an octopus would struggle to untangle it)

tomer-shvadron committed:
fix(plugin): update business type condition in credit check plugin

- Change condition from 'individual' to 'sole_proprietorship'
- Ensure accurate processing for specific business types

tomer-shvadron committed:
fix(plugins): enhance response parsing and error handling

- Replace parse with safeParse for response validation
- Include error handling for invalid response format

tomer-shvadron committed:
fix(plugins): update response structure in verification plugins

- Change response spreading from parsedResponse to parsedResponse.data
- Ensure consistency across the Bank Account and Commercial Credit Check plugins

tomer-shvadron committed:
Merge branch 'dev' into bal-3154
